da/da.go

```diff
@@ -162,12 +162,14 @@ func (dac *DAClient) RetrieveBlocks(ctx context.Context, dataLayerHeight uint64)
 			},
 		}
 	}
-	// ids can be nil if there are no blocks at the requested height.
-	if ids == nil {
+
+	// If no blocks are found, return a non-blocking error.
+	if len(ids) == 0 {
```

Comment:
It seems that an uninitialized slice default value is `nil` so `len(nil) == 0` will still evaluate correctly: https://go.dev/play/p/8Eqo2JKOBVZ.

Also, I think since [go-da/proxy](https://github.com/rollkit/go-da/blob/main/proxy/client.go#L70-L75) wraps the call, should always return an initialized slice of length 0 from `idsPB2DA`

Reply:
Ah I see, great point, thanks for clarifying!